src/or_trajopt/planning.py

```diff
@@ -632,3 +647,45 @@ def _WaypointsToTraj(self, robot, waypoints):
         for (i, waypoint) in enumerate(waypoints):
             traj.Insert(i, waypoint)
         return traj
+
+    def _raiseCollisionErrorForPose(self, robot, pose):
+        """ Identify collision for pose and raise error.
+        This will raise error even if no specific collision is found, 
+        so it should be called only when there is no IK solution. 
+        """
+        from openravepy import (IkFilterOptions,
+                                IkParameterization,
+                                IkParameterizationType)
+
+        manipulator = robot.GetActiveManipulator()
+
+        ik_param = IkParameterization(
+            pose, IkParameterizationType.Transform6D)
+
+        ik_solutions = manipulator.FindIKSolutions(
+            ik_param,
+            openravepy.IkFilterOptions.IgnoreSelfCollisions,
+            ikreturn=False,
+            releasegil=True
+            )
+
+        p = openravepy.KinBody.SaveParameters
+        with robot.CreateRobotStateSaver(p.ActiveDOF):
+            self._checkCollisionForIKSolutions(robot, ik_solutions)
+
+        raise PlanningError('No collision-free IK solution.')
```

Comment:
Unless there is a bug we should never hit this point, correct?

If so, you should change to a generic Python exception class (like `Exception`) so it is not caught (and potentially hidden) by a `MetaPlanner`.